client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js

```diff
@@ -70,6 +65,23 @@ class PaymentMethodStripe extends Component {
 		};
 	}
 
+	componentDidMount() {
+		const { site } = this.props;
+		if ( site && site.ID ) {
+			this.props.fetchAccountDetails( site.ID );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		const { site } = this.props;
+		const newSiteId = ( newProps.site && newProps.site.ID ) || null;
+		const oldSiteId = ( site && site.ID ) || null;
+
+		if ( oldSiteId !== newSiteId ) {
```

Comment:
Do we need to check that `newSiteId` is not `null` here otherwise `fetchAccountDetails` could be called with `null` ( which I guess it probably handles okay )